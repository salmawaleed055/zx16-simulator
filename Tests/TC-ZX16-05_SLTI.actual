Loaded 8 bytes into memory
0x0000: F671    xori x1, -5
0x0002: FC79    li x1, -2
0x0004: FE49    slti x1, -1
0x0006: FFC7    Unknown opcode

--- Final State ---
x0: 0x0000
x1: 0x0001
x2: 0x0000
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0000
a1: 0x0000
PC: 0x0006
---------------------

Simulation finished.
