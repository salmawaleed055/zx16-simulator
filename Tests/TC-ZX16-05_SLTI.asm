# TC-ZX16-05_SLTI.s
# Test Objective: Validate I-type SLTI (Set Less Than Immediate Signed) instruction
# x1 = (-2 < -1) ? 1 : 0 = 1
li x1, -2 # x1 = 0xFFFE
slti x1, -1 # x1 = (int16_t)0xFFFE < (int16_t)0xFFFF ? 1 : 0 => 1
ecall 0x3FF