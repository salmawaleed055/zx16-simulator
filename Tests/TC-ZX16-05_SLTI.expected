0x0000: FC79    li x1, 65534
0x0002: FE49    slti x1, -1
0x0004: FFC7    ecall 0x3FF

--- Final State ---
x0: 0x0000
x1: 0x0001
x2: 0x0000
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0000
a1: 0x0000
PC: 0x0006
---------------------
Simulation finished.